tests/unit/sources/test_sources.py

```diff
@@ -165,3 +165,31 @@ def test_setup_multiple_calls(self, setup_mock):
         self.source.ensure_source_setup()
         self.assertTrue(self.source.is_setup())
         setup_mock.assert_called_once()
+
+
+class TestSourceTearDown(TestCase):
+    """Test that teardown functionality for Source class."""
+    def setUp(self):
+        self.source = Source()
+
+    def test_default_down_false(self):
+        """Test that is_down return False by default."""
+        self.assertFalse(self.source.is_down())
+
+    @patch.object(Source, 'teardown')
+    def test_teardown(self, teardown_mock):
+        """Test that ensure_teardown call teardown and sets the right value
+        for _down attribute."""
+        self.source.ensure_teardown()
+        self.assertTrue(self.source.is_down())
+        teardown_mock.assert_called_once()
```

Comment:
Oh boy, you never mock the object you are testing. The 'Source' you have here is not the real deal, it is a mock created by \@path to look like a Source. If a function from your object calls another of its functions, you have to tests the effects of that call.

Reply:
Just to clarify, this does really mock the whole Source class, just the teardown method to test the ensure_teardown method. That said, you're correct that here neither the teardown nor the setup in the TestSourceSetup class need to mocked. I mocked the setup method in another PR to be able to check if it was called internally, but I see your point that it's not necessary

Reply:
Should we modify both setup and teardown then? @cescgina

Reply:
Yes, let's remove the mocks for both setup and teardown please, given that they are not really needed

Reply:
I've removed them. Thanks to both.